draft-kinnear-webtransport-http2.md

```diff
@@ -63,459 +78,248 @@ Discussion of this draft takes place on the WebTransport mailing list
++-+-------------------------------------------------------------+
+|                            Data (*)                         ...
++---------------------------------------------------------------+
+|                           Padding (*)                       ...
++---------------------------------------------------------------+
 ~~~
-[[ From Client ]]                   [[ From Server ]]
+{: #fig-wt_datagram title="WT_STREAM Frame Format"}
 
-SETTINGS
-SETTINGS_ENABLE_CONNECT_[..] = 1
-SETTINGS_ENABLE_WEBTRANSPORT = 1
+The WT_DATAGRAM frame define the following fields:
 
-                                    SETTINGS
-                                    SETTINGS_ENABLE_CONNECT_[..] = 1
-                                    SETTINGS_ENABLE_WEBTRANSPORT = 1
+   Pad Length:  An 8-bit field containing the length of the frame
+      padding in units of octets.  This field is conditional (as
+      signified by a "?" in the diagram) and is only present if the
+      PADDED flag is set.
 
-HEADERS + END_HEADERS
-+ STREAM_ID = 3
-:method = CONNECT
-:protocol = webtransport
-:scheme = https
-:path = /
-:authority = server.example.com
+   Session ID:  An unsigned 31-bit integer that identifies the
+      stream Connect Stream for this Web Transport stream.  The
+      Session ID MUST be MUST be an open stream negotiated via the
+      extended CONNECT protocol with a `:protocol` value of
+      "webtransport".
 
-                                    HEADERS + END_HEADERS
-                                    + STREAM_ID = 3
-                                    :status = 200
+   Data:  Application data.  The amount of data is the remainder of the
+      frame payload after subtracting the length of the other fields
+      that are present.
 
-                                    WTHEADERS + END_HEADERS
-                                    + STREAM_ID = 2
-                                    + CONNECT_STREAM = 3
-                                    :method = GET
-                                    :scheme = https
-                                    :path = /
-                                    :authority = client.example.com
+The WT_DATAGRAM frame defines the following flags:
 
-WTHEADERS + END_HEADERS
-+ STREAM_ID = 2
-+ CONNECT_STREAM = 3
-:status = 200
+   UNIDIRECTIONAL (0x1):  When set, the stream begins in the
+      "half-closed (remote)" state.
 
-                                    DATA + STREAM_ID = 2
-                                    WebTransport Data
+The recipient MAY respond with a stream error of type WT_STREAM_ERROR if the
+specified WebTransport Connect Stream does not exist, is not a stream
+established via extended CONNECT to use the "webtransport" protocol, or if it is
+in the "closed" or "half-closed (remote)" stream state.
 
-DATA + STREAM_ID = 2 + END_STREAM
-WebTransport Data
+The data in WT_DATAGRAM frames is not subject to flow control.  The receiver MAY
+discard this data if it does not have sufficient space to buffer it.
```

Comment:
Probably worth filing an issue to either reference the appropriate text or just import it about how endpoints should be careful about what they throw out

Reply:
Where is that text now?

Reply:
Potentially in QUIC datagram or H3 datagram itself. Let's just make an issue for this one at the moment and we can dig it up later